### PR TITLE
BUG: algos.rank with readonly values

### PR DESCRIPTION
**Pull request by jbrockmendel**
One of the two current npdev failures is caused by https://github.com/numpy/numpy/pull/17884

### CODE REVIEW

File: pandas/_libs/algos.pyx [hunk 1]
```diff
@@ -824,7 +824,7 @@ def rank_1d(
         if values.dtype != np.object_:
             values = values.astype('O')
     else:
-        values = np.asarray(in_arr)
+        values = np.asarray(in_arr).copy()
```
Comment (jreback) [inline pandas/_libs/algos.pyx:827]: prob can remove L838-841 as we are always copying no?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/algos.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/algos.pyx:827]: good idea, updated

**Comment by jbrockmendel** [comment]
updated + green, this will fix the npdev CI build

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
great

**Comment by simonjayhawkins** [comment]
@meeseeksdev backport 1.2.x

**Pull request merged**